pkg/plugins/secret/dev/dev_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secret_service_test
+
+import (
+	"os"
+
+	"github.com/nitric-dev/membrane/pkg/plugins/secret"
+	secretPlugin "github.com/nitric-dev/membrane/pkg/plugins/secret/dev"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Dev Secret Manager", func() {
+	os.Setenv("LOCAL_SEC_DIR", "./.nitric/")
+
+	AfterSuite(func() {
+		// Cleanup default secrect directory
+		os.RemoveAll("./.nitric/")
+	})
+
+	testSecret := secret.Secret{
+		Name: "Test",
+	}
+	testSecretVal := []byte("Super Secret Message")
+	When("Put", func() {
+		When("Putting a secret to a non-existent secret", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should successfully store a secret", func() {
+				response, err := secretPlugin.Put(&testSecret, testSecretVal)
+				By("Not returning an error")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Returning a non nil response")
+				Expect(response).ShouldNot(BeNil())
+			})
+		})
+		When("Putting a secret to an existing secret", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should succesfully store a secret", func() {
+				response, err := secretPlugin.Put(&testSecret, testSecretVal)
+				By("Not returning an error")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Returning a non nil response")
+				Expect(response).ShouldNot(BeNil())
+			})
+		})
+		When("Putting a secret with an empty name", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should throw an error", func() {
+				emptySecretName := &secret.Secret{}
+				response, err := secretPlugin.Put(emptySecretName, testSecretVal)
+				By("Returning an error")
+				Expect(err).Should(HaveOccurred())
+
+				By("Returning a nil response")
+				Expect(response).Should(BeNil())
+			})
+		})
+	})
```

Comment:
Add invalid secret name.

Reply:
Have raised: https://github.com/nitrictech/membrane/issues/114 to address this in common code.